Lib/tempfile.py

```diff
@@ -98,7 +98,7 @@ def _infer_return_type(*args):
     """Look at the type of all args and divine their implied return type."""
     return_type = None
     for arg in args:
-        if arg is None:
+        if arg is None or isinstance(arg, _os.PathLike):
```

Comment:
Why is this necessary? Can't you simply call `_os.fspath(arg)` just before `if isinstance(arg, bytes)`?